// @sanity/vision/lib/components/QueryErrorDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function QueryErrorDetails(props) {
  var details = props.error.details;

  if (!details || !details.line) {
    return null;
  }

  return _react.default.createElement("div", null, _react.default.createElement("pre", null, _react.default.createElement("code", null, `${details.line}\n${dashLine(details.column)}`)), _react.default.createElement("pre", null, _react.default.createElement("code", null, `Line:   ${details.lineNumber}\nColumn: ${details.column}`)));
}

function dashLine(ln) {
  var line = new Array(ln + 1).join('-');
  return `${line}^`;
}

QueryErrorDetails.propTypes = {
  error: _propTypes.default.shape({
    details: _propTypes.default.shape({
      line: _propTypes.default.string,
      lineNumber: _propTypes.default.number,
      column: _propTypes.default.column
    })
  })
};
var _default = QueryErrorDetails;
exports.default = _default;
